code/modules/projectiles/projectile/beams.dm

```diff
@@ -55,20 +62,32 @@
 		impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser/wall
 
 /obj/projectile/beam/weak
+	tracer_type = /obj/effect/projectile/tracer/laser
+	muzzle_type = /obj/effect/projectile/muzzle/laser
+	impact_type = /obj/effect/projectile/impact/laser
```

Comment:
these vars should be on basic /obj/projectile/beam instead of every subtype overriding it